index.bs

```diff
@@ -2429,6 +2438,7 @@ optionally evidence of [=user consent=] to a specific transaction.
     interface AuthenticatorAssertionResponse : AuthenticatorResponse {
         [SameObject] readonly attribute ArrayBuffer      authenticatorData;
         [SameObject] readonly attribute ArrayBuffer      signature;
+        [SameObject] readonly attribute DOMString?       transport;
```

Comment:
I think we should at least consider also having this field in the _AuthenticatorAttestationResponse_
I'd like to see that idea discussed on a future WebAuthn call before merging this PR.

My thinking here is that getTransports(), which is a method that is also part of the  _AuthenticatorAttestationResponse_ interface, returns a list and in cases where the authenticator supports more than one transport it is still useful to know which transport was used during a registration ceremony.

Reply:
In that case, perhaps the property should be on `PublicKeyCredential` instead of on the `AuthenticatorResponse`? That would also help separate between `transport` and `transports` fields - having both on the same object seems like asking for bugs to happen.

Reply:
Yes - that would be a good place for it.

Reply:
Yup, I agree, having both transport and transports fields could get confusing, which is why instead of exposing the actual transport, it'll now be exposed as platform attachment, which is either platform or cross-platform as suggested by Adam above. Does that resolve the issue with having similar names?

Reply:
I would like to see and review the new intended PR as it's not quite clear to me what is now being proposed for API changes.

Reply:
This also relates to a discussion in https://github.com/w3c/webauthn/pull/1621#discussion_r709721573 .

Reply:
Keeping `transport`/`attachment`/etc... on `PublicKeyCredential` would also help developers track where that's returned (_if_ it gets returned, anyway), since it'd be a top-level property in the same location no matter if it's a registration or authentication response:

**Registration:**
```js
{
  id: '...',
  rawId: '...',
  response: {
    attestationObject: '...',
    clientDataJSON: '...',
  },
  type: 'public-key',
  transports: ['usb', '...'],  // <---
}
```

**Authentication:**
```js
{
  id: '...',
  rawId: '...',
  response: {
    authenticatorData: '...',
    clientDataJSON: '...',
    signature:
      '...',
  },
  type: 'public-key',
  transport: 'internal',   // <---
  attachment: 'platform',  // <---
}
```

Reply:
@sbweeden, I've made updates to the API, have a look when you can.

also, @MasterKale, you're suggesting to move transports in attestation response during registration to PublicKeyCredential out of response? The transports fit more inside the attestationResponse, and there'll be only an attachment field (not transport & attachment) for both Registration & Authentication.

Reply:
Forgive me, I mis-remembered something I do in one of my libraries as being part of the spec. I forgot that transports are only available as `AuthenticatorAttestationResponse.getTransports()`. I am _not_ advocating for moving anything out of `AuthenticatorAttestationResponse`.

What I _am_ pushing for is something like `AuthenticatorAssertionResponse.getAttachment()`, to keep things consistent with `AuthenticatorAttestationResponse.getTransports()`. I'm trying to keep track of everything being discussed in this PR and if my understanding is correct I think I agree with @sbweeden and disagree with @emlun...

> I would like to see and review the new intended PR as it's not quite clear to me what is now being proposed for API changes.

I think I agree with @sbweeden on this one, if this PR is moving away from exposing the exact transport used during authentication for authenticator attachment instead then it might make sense to close this out and prep a new PR.

Reply:
I think the idea now is to add the attachment property to both registration and authentication responses, since `transports` communicates a different (though related) piece of information. That would be the reason to put it on `PublicKeyCredential` instead, if it's going to be on both response types anyway.

Reply:
@MasterKale, I agree with @emlun,I'm not seeing any reason for consistency between AuthenticatorAttestationResponse.getTransports() & AuthenticatorAssertionResponse.getAttachment(). They have different use cases, and for the main reason the attachment will be used, I'm pushing to keep the attachment on the top level of PublicKey Credential so its exposed both during registration (attestation) as well as authentication (assertion)

> I think I agree with @sbweeden on this one, if this PR is moving away from exposing the exact transport used during authentication for authenticator attachment instead then it might make sense to close this out and prep a new PR.

This PR has been updated with the change from transport -> authenticator attachment, and has good discussions within it, so I believe i'm going to keep this PR and not open up a new one.

Reply:
Alright, I'm all caught up after re-reviewing the latest content of the PR. I'm onboard now with putting it on PublicKeyCredential :v: